models/issue_tracked_time.go

```diff
@@ -131,3 +186,111 @@ func TotalTimes(options FindTrackedTimesOptions) (map[*User]string, error) {
 	}
 	return totalTimes, nil
 }
+
+// DeleteIssueUserTimes deletes times for issue
+func DeleteIssueUserTimes(issue *Issue, user *User) error {
+	opts := FindTrackedTimesOptions{
+		IssueID: issue.ID,
+		UserID:  user.ID,
+	}
+
+	removedTime, err := GetTrackedSeconds(opts)
+	if err != nil {
+		return err
+	}
+	if removedTime == 0 {
+		return ErrNotExist{}
+	}
+
+	removedTime, err = deleteTimes(opts)
+
+	if err := issue.loadRepo(x); err != nil {
+		return err
+	}
+	if _, err := CreateComment(&CreateCommentOptions{
+		Issue:   issue,
+		Repo:    issue.Repo,
+		Doer:    user,
+		Content: "- " + SecToTime(removedTime),
+		Type:    CommentTypeDeleteTimeManual,
+	}); err != nil {
+		return err
+	}
+	return err
+}
+
+func deleteTimes(opts FindTrackedTimesOptions) (removedTime int64, err error) {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err = sess.Begin(); err != nil {
+		return
+	}
+
+	tt, err := GetTrackedTimes(opts)
```

Comment:
`getTrackedTimes(e, opts)`

Reply:
@lunny do not exist

Reply:
@6543 then you have to create one.

Reply:
I'd say we should just delete the times here - why bother getting them out of the db at all?

You have calculated their summed time already so use that.

Reply:
With of course the caveat that we need to lock the table somehow

Reply:
do 9661f85b465787d6f1d4452706021d73f8a1165f resolve this?

Reply:
@zeripath